packages/experience-builder-sdk/src/core/componentRegistry.ts

```diff
@@ -189,6 +192,115 @@ export const runRegisteredComponentValidations = () => {
+  element: HTMLDivElement,
+  cssVariableValue: string,
+  cssAttribute: any,
+) => {
+  element.style[cssAttribute] = cssVariableValue;
+  const styles = getComputedStyle(element);
+  const resolvedValue = styles.getPropertyValue(cssAttribute);
+  return resolvedValue;
+};
+
+const getAllCssVariableValues = (
+  element: HTMLDivElement,
+  cssVariable: Record<string, string>,
+  cssAttribute: any,
+) => {
+  const resolvedCssVariables = {} as Record<string, string>;
+
+  Object.keys(cssVariable).forEach((key) => {
+    const cssVariableValue = cssVariable[key];
+    if (CssVarRegex.test(cssVariableValue)) {
+      const resolvedValue = getSingleCssVariableValue(element, cssVariableValue, cssAttribute);
+      resolvedCssVariables[cssVariableValue] = resolvedValue;
+    }
+  });
+  return resolvedCssVariables;
+};
+
+const resolveCssVariables = (designTokensDefinition: DesignTokensDefinition) => {
+  const {
+    spacing,
+    sizing,
+    color,
+    borderRadius,
+    fontSize,
+    lineHeight,
+    letterSpacing,
+    textColor,
+    border,
+  } = designTokensDefinition;
+  const resolvedCssVariables = {} as Record<string, string>;
+
+  // Create an element
+  const element = document.createElement('div');
+  document.body.appendChild(element);
+
+  if (spacing) {
+    // Doesn't matter if it's margin or padding or gap related attributes. You just need to resolve the css variables related to the spacing design token once
+    // since each dropdown for those individual css attributes will have the same design token values
+    const rawResolvedValues = getAllCssVariableValues(
+      element,
+      spacing as Record<string, string>,
+      'margin',
+    );
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
+  if (sizing) {
+    const rawResolvedValues = getAllCssVariableValues(element, sizing, 'width');
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
+  if (color) {
+    const rawResolvedValues = getAllCssVariableValues(element, color, 'background-color');
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
+  if (borderRadius) {
+    const rawResolvedValues = getAllCssVariableValues(element, borderRadius, 'border-radius');
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
+  if (fontSize) {
+    const rawResolvedValues = getAllCssVariableValues(element, fontSize, 'font-size');
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
+  if (lineHeight) {
+    const rawResolvedValues = getAllCssVariableValues(element, lineHeight, 'line-height');
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
+  if (letterSpacing) {
+    const rawResolvedValues = getAllCssVariableValues(element, letterSpacing, 'letter-spacing');
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
+  if (textColor) {
+    const rawResolvedValues = getAllCssVariableValues(element, textColor, 'color');
+    Object.assign(resolvedCssVariables, rawResolvedValues);
+  }
```

Comment:
small nit - this could be refactored to use a loop to reduce repetition. for example:
```ts
const cssProperties = [
  { variable: sizing, property: 'width' },
  { variable: color, property: 'background-color' },
  { variable: borderRadius, property: 'border-radius' },
  { variable: fontSize, property: 'font-size' },
  { variable: lineHeight, property: 'line-height' },
  { variable: letterSpacing, property: 'letter-spacing' },
  { variable: textColor, property: 'color' },
];

cssProperties.forEach(({ variable, property }) => {
  if (variable) {
    const rawResolvedValues = getAllCssVariableValues(element, variable, property);
    Object.assign(resolvedCssVariables, rawResolvedValues);
  }
});
```